### DISCUSSION TITLE
What is differece between goBack() and pop() in router

### DISCUSSION THREAD

**Discussion by ibcylon** [johnpatrickmorgan/FlowStacks#74] [Q&A] [answered]
The content is a same with title.

I know `pop()` can be used only pushed View, but I wonder the reason separate them.

would you explain the useCase it must use `goBack` or more make sense to use `goBack` than `pop()`?

Thank you for making flowStacks

**Comment by johnpatrickmorgan** [comment, marked as answer]
Hi @ibcylon, thanks for your question. You're right that `goBack` will go back to the previous screen, regardless of whether the current screen was pushed or presented. And `pop` does the same thing, but in debug builds it will stop program execution if the current screen was presented rather than pushed. This might be useful if you want to get an early warning while developing that your assumptions about the navigation state are not correct. 

So it's a matter of personal preference: if you want to verify your assumption that the current screen has been pushed, use `pop`; otherwise, use `goBack` as it's more flexible.
